maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java

```diff
@@ -129,14 +132,23 @@ private void multiThreadedProjectTaskSegmentBuild( ConcurrencyDependencyGraph an
                                                        Map<MavenProject, ProjectSegment> projectBuildList,
                                                        ThreadOutputMuxer muxer )
     {
+        // gather artifactIds which are not unique so that the respective thread names can be extended with the groupId
+        Set<String> duplicateArtifactIds = projectBuildList.keySet().stream()
```

Comment:
1. I wonder how much calculation overhead this poses to the execution
2. This won't work for 3.8.x because of Java 8 features.

Reply:
> 1. I wonder how much calculation overhead this poses to the execution

Even for hunderts of projects this should be alright, IMO, but I haven't done any measurements.
We could also switch over to something like: https://stackoverflow.com/a/30741906/9529981

> 2. This won't work for 3.8.x because of Java 8 features.

Right. I can create a Java 8 compatible backport PR if you want.

Reply:
> > 1. I wonder how much calculation overhead this poses to the execution

This is done only once afaik, and is limited by the number of projects so the computational cost remains low.  I don't see any potential problem here.
 
> > 2. This won't work for 3.8.x because of Java 8 features.
> 
> Right. I can create a Java 8 compatible backport PR if you want.

Please do.